presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -361,25 +362,31 @@ private static boolean process(QueryRunner queryRunner, String sql, OutputFormat
 
     private static MemoryHistory getHistory()
     {
-        MemoryHistory history;
-        File historyFile = new File(getUserHome(), ".presto_history");
+        return getHistory(new File(getUserHome(), ".presto_history"));
+    }
+
+    @VisibleForTesting
+    static MemoryHistory getHistory(File historyFile)
+    {
         if (!historyFile.canWrite() || !historyFile.canRead()) {
             System.err.printf("WARNING: Failed to load history file (%s): Not readable/writable path (%s). " +
                             "History will not be available during this session.%n",
                     historyFile, historyFile.getAbsolutePath());
-            history = new MemoryHistory();
+            MemoryHistory history = new MemoryHistory();
+            history.setAutoTrim(true);
+            return history;
         }
-        else {
-            try {
-                history = new FileHistory(historyFile);
-                history.setMaxSize(10000);
-            }
-            catch (IOException e) {
-                System.err.printf("WARNING: Failed to load history file (%s): %s. " +
-                                "History will not be available during this session.%n",
-                        historyFile, e.getMessage());
-                history = new MemoryHistory();
-            }
+
+        MemoryHistory history;
+        try {
+            history = new FileHistory(historyFile);
+            history.setMaxSize(10000);
+        }
+        catch (IOException e) {
+            System.err.printf("WARNING: Failed to load history file (%s): %s. " +
```

Comment:
if we can write and read to/from a file then maybe we should not catch this exception.

As I understand this exception can be thrown when file is corrupted.

Reply:
The method declares throwing `IOException` - meaning you have to catch it.
Being able to read/write from it is just on OS level and simply check if the file system will allow it (not checking validity) .